### ISSUE TITLE
Spurious spaces inserted with curly braces

### ISSUE THREAD

**Issue by simonbyrne** [Humans-of-Julia/BibParser.jl#5]
```julia
julia> using BibParser

julia> bibtex = """@article{Heun1900,
          title = {Neue Methoden zur approximativen Integration der
          Differentialgleichungen einer unabh\\\"{a}ngigen Ver\\\"{a}nderlichen},
          author = {Heun, Karl},
          journal = {Z. Math. Phys},
          volume = {45},
          pages = {23--38},
          year = {1900},
       }"""
"@article{Heun1900,\n   title = {Neue Methoden zur approximativen Integration der\n   Differentialgleichungen einer unabh\\\"{a}ngigen Ver\\\"{a}nderlichen},\n   author = {Heun, Karl},\n   journal = {Z. Math. Phys},\n   volume = {45},\n   pages = {23--38},\n   year = {1900},\n}"

julia> _,bib = BibParser.parse_entry(bibtex)
("", OrderedCollections.OrderedDict("Heun1900" => Dict("volume" => "45","author" => "Heun , Karl","pages" => "23--38","journal" => "Z. Math. Phys","year" => "1900","title" => "Neue Methoden zur approximativen Integration der Differentialgleichungen einer unabh\\ \" { a } ngigen Ver\\ \" { a } nderlichen","type" => "article")))

julia> bib["Heun1900"]["title"]
"Neue Methoden zur approximativen Integration der Differentialgleichungen einer unabh\\ \" { a } ngigen Ver\\ \" { a } nderlichen"
```

**Comment by Azzaare** [comment]
I definitely need to make a new parser (instead of borrowing one from someone ><). As mentioned in [https://github.com/ali-ramadhan/DocumenterCitations.jl/pull/21](https://github.com/ali-ramadhan/DocumenterCitations.jl/pull/21), I will do it next month.

I plan to follow the model of [Tokenize.jl](https://github.com/JuliaLang/Tokenize.jl), and integrate LaTeX special character and diacritical marks.

**Comment by Azzaare** [comment]
Spurious spaces troubles should be solved now (v0.2.7)
However, converting common TeX to Unicode is not possible yet. Hopefully, we can make something for the next version.

Closing this.
